src/GraphQL/Utilities/Visitors/Custom/PatternMatchingVisitor.cs

```diff
@@ -37,9 +37,16 @@ public override void VisitInterfaceFieldArgumentDefinition(QueryArgument argumen
     {
         // look for @pattern directive applied to the field argument or input field, and if found, use Validate to set the validation method
         var applied = fieldArgOrInputField.FindAppliedDirective("pattern");
-        if (applied?.FindArgument("regex")?.Value is not string regex)
+        var regexArgument = applied?.FindArgument("regex");
+        if (regexArgument == null)
             return null;
 
+        if (regexArgument.Value == null)
+            throw new ArgumentException("Pattern directive 'regex' argument must have non-null value");
```

Comment:
Do you want to include the location as well?

Reply:
Should be easy if you pass in a few more args into `GetValidator` I think.

Reply:
This throws during schema validation so a location would probably be valuable.

Reply:
If we wanted to get fancier we could catch and rewrap exceptions caught in the regex constructor (in case of invalid patterns) to add the location as well.

Reply:
How do I do that?

Reply:
I'm thinking something like this:

Change line 36 & 37 to:

```cs
private static Action<object?>? GetValidator(QueryArgument? argument, FieldType field, IInterfaceGraphType type)
{
    var fieldArgOrInputField = argument ?? field;
```

Then change the exception message to:

```cs
throw new ArgumentException($"Pattern directive 'regex' argument at {GetLocation()} must be of 'string' type");
```

with a local method

```cs
string GetLocation()
    => argument != null
        ? $"{type.Name}.{field.Name}.{argument.Name}"
        : $"{type.Name}.{field.Name}";
```